### PR TITLE
dotool: init at 1.3

### LINKED ISSUE #229625: Package request: dotool
**Project description**

Perhaps it would also require a module to set the correct permission for /dev/uinput?

---

dotool reads commands from stdin and simulates keyboard and pointer events. It works everywhere on Linux, including in X11, Wayland and TTYs.

**Metadata**

* homepage URL: https://git.sr.ht/~geb/dotool
* source URL: https://git.sr.ht/~geb/dotool/
* license: GPLv3
* platforms: unix, linux

## Permission

dotool requires permission to `/dev/uinput` to create the virtual input
devices, and a udev rule grants this to users in group input.

You could try:

    echo type hello | dotool

and if need be, you can run:

    sudo groupadd -f input
    sudo usermod -a -G input $USER

and re-login and trigger the udev rule or just reboot.

## Usage

See `dotool --help`, but this greets the world:

    echo 'type Sup, Lads!' | dotool

and this screams for three seconds:

    { echo keydown A; sleep 3; echo key H shift+1; } | dotool

There is an initial delay registering the virtual devices, but you can
keep writing commands to the same instance or use the daemon and client,
`dotoold` and `dotoolc`.

    dotoold &
    echo type super | dotoolc
    echo type speedy | dotoolc

## Keyboard Layouts

dotool will produce gobbledygook if your environment has assigned it a
different keyboard layout than it's simulating keycodes for.  You can
match them up with the environment variables `DOTOOL_XKB_LAYOUT` and
`DOTOOL_XKB_VARIANT`.

    echo type azerty | DOTOOL_XKB_LAYOUT=fr dotool

## Numen and Contact

dotool was written for [Numen](https://numenvoice.org), which has a
[chat on Matrix](https://matrix.to/#/#numen:matrix.org) you're welcome to join.

You can also send questions, thoughts or patches by composing an email to
[~geb/public-inbox@lists.sr.ht](https://lists.sr.ht/~geb/public-inbox).

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

Closes  #229625.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nagy** [comment]
I had this packaged in my nur-repository as well https://github.com/nagy/nur-packages/blob/master/pkgs/dotool.nix .

Things you can take from there also would be the description and the changelog url.

**Comment by nagy** [comment]
In the `postInstall` you also install https://git.sr.ht/~geb/dotool/tree/master/item/dotoold and https://git.sr.ht/~geb/dotool/tree/master/item/dotoolc . Both of there are pointing to `/bin/echo` , which does not exist on NixOS. Also at the end `dotoold` refers to `dotool` which may not exist if you run this from nix-shell. 

Maybe we should patch the `/bin/echo` to `echo` and the `dotool` inside of `dotoold` into `$out/bin/dotool`.

**Review by drupol — APPROVED** [review]

**Comment by dit7ya** [comment]
@drupol Please don't merge this one - I want to address @nagy's comments once I have some time.

**Comment by drupol** [comment]
Yes I noticed, this is why I haven't pushed the button yet.

**Comment by drupol** [comment]
@dit7ya What's the status of this PR ?

**Comment by dit7ya** [comment]
I will try to get to it this weekend. If anyone wants to take over the PR, please feel free to.

File: pkgs/tools/dotool/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  src = fetchFromSourcehut {
+    owner = "~geb";
+    repo = "dotool";
+    rev = version;
+    hash = "sha256-z0fQ+qenHjtoriYSD2sOjEvfLVtZcMJbvnjKZFRSsMA=";
+  };
+
+  vendorHash = "sha256-v0uoG9mNaemzhQAiG85RequGjkSllPd4UK2SrLjfm7A=";
+
+  patches = [ ./fix-paths.patch ];
```
Comment (drupol) [inline pkgs/tools/dotool/default.nix:21]: Would it be possible to add a comment for the patch ?

**Review by drupol — APPROVED** [review]

**Pull request merged**